src/middlewared/middlewared/plugins/cache.py

```diff
@@ -26,18 +28,40 @@ def get(self, key):
         Raises:
             KeyError: not found in the cache
         """
-        return self.__cache[key]
+        if isinstance(self.__cache[key].timeout, float):
+            self.get_timeout(key)
 
-    @accepts(Str('key'), Any('value'))
-    def put(self, key, value):
+        return self.__cache[key][0]
+
+    @accepts(Str('key'), Any('value'), Int('timeout', default=0))
+    def put(self, key, value, timeout):
         """
         Put `key` of `value` in the cache.
         """
-        self.__cache[key] = value
+        kv_tuple = namedtuple('Cache', ['value', 'timeout'])
```

Comment:
kv_tuple could be outside in a global namespace, no need to redefine it every `put`

e.g. CacheEntry = namedtuple('Cache', ['value', 'timeout'])